crates/protocol/protocol/src/batch/single.rs

```diff
@@ -155,6 +156,19 @@ impl SingleBatch {
             }
         }
 
+        // If this is the first block in the interop hardfork, and the batch contains any
+        // transactions, it must be dropped.
+        if !cfg.is_interop_active(self.timestamp - cfg.block_time) &&
+            cfg.is_interop_active(self.timestamp) &&
```

Comment:
May be worth adding a `is_interop_activation_block` function? We have this in Go and use it also e.g. in the sequencer for the upgrade block.

Reply:
For sure. I'll ticket that out for a follow up.

Reply:
https://github.com/op-rs/kona/issues/1603